### PR TITLE
added support for macos-latest aarch64

### PR DESCRIPTION
**Pull request by danbugs**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -150,12 +148,16 @@ run-restaurant-backend:
 ### GITHUB RELEASES
 .PHONY: prepare-release
 prepare-release:
-	tar -C target/ -czvf slight-unix.tar.gz release/slight
+	tar -C target/ -czvf slight-ubuntu.tar.gz release/slight
```
Comment (Mossaka) [inline Makefile:151]: shouldn't this be `slight-linux-arm64.tar.gz`?

I would recommend put semantic version in the name.

File: Makefile [hunk 2]
```diff
@@ -150,12 +148,16 @@ run-restaurant-backend:
...
 	tar -C templates/ -czvf rust-template.tar.gz rust
 	tar -C templates/ -czvf c-template.tar.gz c
 
 .PHONY: prepare-release-win
 prepare-release-win:
-	tar -C target/ -czvf slight-win.tar.gz release/slight.exe
+	tar -C target/ -czvf slight-windows.tar.gz release/slight.exe
```
Comment (Mossaka) [inline Makefile:157]: same comment as above. `slight-windows-x86_64`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -49,26 +59,61 @@ jobs:
...
+        run: tar -C target/ -czvf slight-macos-aarch64.tar.gz aarch64-apple-darwin/release/slight
+        if: ${{ fromJSON(matrix.config.os == 'macos-latest') && 
+          fromJSON(matrix.config.arch == 'aarch64') }}
+
+      - name: "add wits, and slight-ubuntu and templates tars to release"
         uses: softprops/action-gh-release@v1
         with:
           files: |
-            slight-unix.tar.gz
+            slight-ubuntu.tar.gz
```
Comment (Mossaka) [inline .github/workflows/release.yml:93]: same comment as below

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -49,26 +59,61 @@ jobs:
...
           OPENSSL_DIR: C:\Program Files\OpenSSL-Win64\
+          OPENSSL_NO_VENDOR: true
+
+      - name: "build for macos amd64"
+        run: make build
+        if: ${{ fromJSON(matrix.config.os == 'macos-latest') && 
+          fromJSON(matrix.config.arch == 'amd64') }}
+      
+      - name: "build for macos aarch64"
+        run: cargo build --manifest-path ./slight/Cargo.toml --release ${{ matrix.config.extraArgs }}
```
Comment (Mossaka) [inline .github/workflows/release.yml:70]: could you please put the command in makefile?

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -88,16 +94,21 @@ jobs:
...
       - name: "Build SpiderLightning and Slight"
         run: make build
-        if: ${{ fromJSON(matrix.config.os == 'macos-latest') }}
-        env:
-          OPENSSL_ROOT_DIR: /usr/local/opt/openssl
+        if: ${{ fromJSON(matrix.config.os == 'macos-latest') && 
+          fromJSON(matrix.config.arch == 'amd64') }}
+
+      - name: "Build SpiderLightning and Slight"
+        run: cargo build --manifest-path ./slight/Cargo.toml --release ${{ matrix.config.extraArgs }}
```
Comment (Mossaka) [inline .github/workflows/ci.yaml:101]: same comment as below

**Review by Mossaka — COMMENTED** [review]
lgtm! had some nitpicking comments

File: Makefile (hunk 1 above)
Comment (danbugs) [reply Makefile]: done ✅

File: Makefile (hunk 2 above)
Comment (danbugs) [reply Makefile]: done ✅

File: .github/workflows/release.yml (hunk 3 above)
Comment (danbugs) [reply .github/workflows/release.yml]: done ✅

File: .github/workflows/ci.yaml (hunk 5 above)
Comment (danbugs) [reply .github/workflows/ci.yaml:101]: ~didn't want to do it 'cause of the github action specific argument

File: .github/workflows/release.yml (hunk 4 above)
Comment (danbugs) [reply .github/workflows/release.yml:70]: ~didn't want to do it 'cause of the action specific argument

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -19,14 +19,18 @@ jobs:
       matrix:
         config:
           - os: "ubuntu-latest"
-            extension: ""
             platformAgnosticChecks: true
-          - os: "macos-latest"
-            extension: ""
-            extraArgs: ""
+          - {
```
Comment (Mossaka) [inline .github/workflows/ci.yaml:23]: stupid question: do we really need curly braces to define a map of key value pairs?

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -71,6 +76,7 @@ jobs:
         run: |
           rustup target add wasm32-wasi
           rustup target add wasm32-unknown-unknown
+          rustup target add aarch64-apple-darwin
```
Comment (Mossaka) [inline .github/workflows/ci.yaml:79]: nitpicking: this target is only needed in macos worker.

File: .github/workflows/ci.yaml (hunk 5 above)
Comment (Mossaka) [reply .github/workflows/ci.yaml:101]: sounds reasonable. My only concern is that if we refactor the code in the future that requires us to change the `cargo build` command to some other path, we might forget about changing the command in the github action.

**Review by Mossaka — APPROVED** [review]

**Pull request merged**